lighthouse-core/gather/computed/page-dependency-graph.js

```diff
@@ -34,52 +38,203 @@ class PageDependencyGraphArtifact extends ComputedArtifact {
+        if (distance > 0 && distance < minDistance) {
+          minCandidate = candidate;
+          minDistance = distance;
+        }
+      });
+
+      if (!minCandidate) return;
+      cpuNode.addDependency(minCandidate);
+    }
+
+    const timers = new Map();
+    for (const node of cpuNodes) {
+      for (const evt of node.children) {
+        if (!evt.args.data) continue;
+
+        const url = evt.args.data.url;
+        const stackTraceUrls = (evt.args.data.stackTrace || []).map(l => l.url).filter(Boolean);
+
+        switch (evt.name) {
+          case 'TimerInstall':
+            timers.set(evt.args.data.timerId, node);
+            stackTraceUrls.forEach(url => addDependencyOnUrl(node, url));
+            break;
+          case 'TimerFire': {
+            const installer = timers.get(evt.args.data.timerId);
+            if (!installer) break;
+            installer.addDependent(node);
+            break;
+          }
+
+          case 'InvalidateLayout':
+          case 'ScheduleStyleRecalculation':
+            stackTraceUrls.forEach(url => addDependencyOnUrl(node, url));
+            break;
+
+          case 'EvaluateScript':
+            addDependencyOnUrl(node, url);
+            stackTraceUrls.forEach(url => addDependencyOnUrl(node, url));
+            break;
+
+          case 'XHRReadyStateChange':
+            // Only create the dependency if the request was completed
+            if (evt.args.data.readyState !== 4) break;
+
+            addDependencyOnUrl(node, url);
+            stackTraceUrls.forEach(url => addDependencyOnUrl(node, url));
+            break;
+
+          case 'FunctionCall':
+          case 'v8.compile':
+            addDependencyOnUrl(node, url);
+            break;
+
+          case 'ParseAuthorStyleSheet':
+            addDependencyOnUrl(node, evt.args.data.styleSheetUrl);
+            break;
+
+          case 'ResourceSendRequest':
+            addDependentNetworkRequest(node, evt.args.data.requestId, evt);
+            stackTraceUrls.forEach(url => addDependencyOnUrl(node, url));
+            break;
+        }
+      }
+
+      if (node.getNumberOfDependencies() === 0) {
+        node.addDependency(rootNode);
+      }
+    }
+  }
+
+  /**
+   * @param {!TraceOfTabArtifact} traceOfTab
+   * @param {!Array<!WebInspector.NetworkRequest>} networkRecords
+   * @return {!Node}
+   */
+  static createGraph(traceOfTab, networkRecords) {
+    const networkNodeOutput = this.getNetworkNodes(networkRecords);
+    const cpuNodes = this.getCPUNodes(traceOfTab);
+
+    const rootRequest = networkRecords.reduce((min, r) => (min.startTime < r.startTime ? min : r));
+    const rootNode = networkNodeOutput.idToNodeMap.get(rootRequest.requestId);
+
+    this.linkNetworkNodes(rootNode, networkNodeOutput, networkRecords);
+    this.linkCPUNodes(rootNode, networkNodeOutput, cpuNodes, traceOfTab);
 
     return rootNode;
   }
 
   /**
    * @param {!Node} rootNode
-   * @return {number}
+   * @return {}
```

Comment:
yeah along with the other host of refactors for post #3163, estimator is just wrong at this point, simulator makes a lot more sense based on what it's evolved into since the time is easily achieved from the node information